packages/liveblocks-react/src/umbrella-store.ts

```diff
@@ -610,20 +612,32 @@ export class UmbrellaStore<M extends BaseMetadata> {
         );
       }
 
-      const result = await client.getInboxNotifications({ cursor });
+      try {
+        const result = await client.getInboxNotifications({ cursor });
 
-      this.updateThreadsAndNotifications(
-        result.threads as ThreadData<M>[], // TODO: Figure out how to remove this casting
-        result.inboxNotifications
-      );
+        this.updateThreadsAndNotifications(
+          result.threads as ThreadData<M>[], // TODO: Figure out how to remove this casting
+          result.inboxNotifications
+        );
 
-      // We initialize the `_lastRequestedNotificationsAt` date using the server timestamp after we've loaded the first page of inbox notifications.
-      if (this._notificationsLastRequestedAt === null) {
-        this._notificationsLastRequestedAt = result.requestedAt;
-      }
+        // We initialize the `_lastRequestedNotificationsAt` date using the server timestamp after we've loaded the first page of inbox notifications.
+        if (this._notificationsLastRequestedAt === null) {
+          this._notificationsLastRequestedAt = result.requestedAt;
+        }
+
+        const nextCursor = result.nextCursor;
+        return nextCursor;
+      } catch (err) {
+        // If the error is a 403 Forbidden error, we do not want to keep retrying
+        if (err instanceof NotificationsApiError && err.status === 403) {
```

Comment:
I think we'll want to extend this to any error in the 4xx range, not just the 403. What 4xx means is: you had no access and without changing the request you just made retrying will not have an effect.

4xx: Request refused for a known reason, you need to make a different request
5xx: Request refused for an unknown/unexpected reason, please retry